tests/cupy_tests/core_tests/test_reduction.py

```diff
@@ -189,6 +189,47 @@ def test_invalid_kernel_name(self):
                 'T x', 'T y', 'x', 'a + b', 'y = a', '0', name='1')
 
 
+class AbstractReductionTestBaseMultipleArguments:
+
+    def get_sum_func(self):
+        raise NotImplementedError()
+
+    def check_sign(a):
+        i = -1 if a < 0 else 1
+        return i
+
+    @testing.numpy_cupy_allclose(contiguous_check=False)
+    def check_sum(self, shape, xp, axis=None, keepdims=False, trans=None):
+        a = testing.shaped_random(shape, xp, 'b')
+        if trans:
+            a = a.tanspose(*trans)
+        sum_func = self.get_sum_func()
+        if xp == cupy:
+            k = sum_func(a, axis, keepdims=keepdims)
+            s = check_sign(k)
+            return k, s
+        else:
+            return a.sum(axis=axis, keepdims=keepdims, dtype='b')
+
+
+class ReductionKernelMultipleArgumentsTestBase(
+        AbstractReductionTestBaseMultipleArguments):
+
+    def get_sum_func(self):
+        return cupy.ReductionKernel(
+            'T x', 'T out', 'x', 'a + b', 'out = a, b', '0', 'my_sum')
```

Comment:
This time I think of making ReductionKernel able to handle multiple values only in the post-reduction phase (Is it also okay for you?). So would you try to make the following ReductionKernel pass the test? The ReductionKernel  takes two arrays for output, `out0` and `out1`, then assigning them to sum of `x` as well as its negated sum. The negated sum does not have much sense, just testing purpose.

```
return cupy.ReductionKernel(
            'T x', 'T out0, T out1', 'x', 'a + b', 'out0 = a; out1 = -a', '0', 'my_sum_neg_sum')
```